targets/TARGET_NUVOTON/TARGET_M2354/TARGET_TFM/TARGET_NU_M2354/COMPONENT_TFM_S_FW/README.md

```diff
@@ -0,0 +1,132 @@
+```sh
+git clone https://github.com/OpenNuvoton/trusted-firmware-m -b nuvoton_mbed_m2354_tfm-1.2
+```
+
+## Customizing TF-M
+
+TF-M is out of scope of Mbed.
+We only provide limited support for customizing M2354 port in TF-M.
+
+### Default hardware partition
+
+In TF-M, by default, the M2354 hardware is partitioned as follows:
+
+- Flash (1024KiB in total): 448KiB for secure and 576KiB for nonsecure.
+- SRAM (256KiB in total): 80KiB for secure and 176KiB for nonsecure.
+- Peripherals: Most are configured to nonsecure except the following, which are hardwired or reserved:
+    - **SYS/CLK**: Hardwired to secure. Accessible to Mbed through secure gateway.
+    - **FMC**: Hardwired to secure. Inaccessible to Mbed.
+    - **WDT**: Hardwired to secure. Inaccessible to Mbed.
+    - **RTC**: Configured to secure. Accessible to Mbed through secure gateway.
+    - **TMR0/1**: Hardwired to secure. Inaccessible to Mbed.
+    - **TMR2/3**: Configured to nonsecure. TMR2 is used for TF-M NSPE test when enabled.
+    - **TMR4/5**: Configured to nonsecure for Mbed `us_ticker`/`lp_ticker`.
+    - **PDMA0**: Hardwired to secure. Inaccessible to Mbed.
+    - **PDMA1**: Configured to nonsecure for Mbed asynchronous transfer.
+    - **CRYPTO**: Configured to secure. Inaccessible to Mbed.
+    - **TRNG**: Hardwired to secure. Accessible to Mbed through secure gateway.
+
+### Defining Flash for TF-M/Mbed
+
+To define memory spec of Flash for TF-M/Mbed, search/change the line:
+
+**trusted-firmware-m/platform/ext/target/nuvoton/m2354/partition/flash_layout.h**:
+```C
+/* Max Flash size for TF-M + bootloader information */
+#define FLASH_S_PARTITION_SIZE          (0x50000)
+/* Max Flash size for Mbed + bootloader information */
+#define FLASH_NS_PARTITION_SIZE         (0x90000)
+```
+
+M2354 has 1024KiB Flash in total, among which 128KiB have been allocated for bootloader code and ITS/PS storage.
+896KiB are left for `FLASH_S_PARTITION_SIZE` and `FLASH_NS_PARTITION_SIZE`.
+
+**NOTE**: `FLASH_S_PARTITION_SIZE` and `FLASH_NS_PARTITION_SIZE` must be sector size (2KiB)-aligned.
+
+### Defining SRAM for TF-M/Mbed
+
+To define memory spec of SRAM for TF-M/Mbed, search/change the line:
+
+**trusted-firmware-m/platform/ext/target/nuvoton/m2354/partition/region_defs.h**:
+```C
+/* Max SRAM size for TF-M */
+#define S_DATA_SIZE     (80 * 1024)
+/* Max SRAM size for Mbed = Total - Max SRAM size for TF-M */
+#define NS_DATA_SIZE    (TOTAL_RAM_SIZE - S_DATA_SIZE)
+```
+
+**NOTE**: `S_DATA_SIZE` and `NS_DATA_SIZE` must be 16KiB-aligned required by M2354 Security Configuration Unit (SCU).
+
+## Rebuilding TF-M
+
+### Environment
+
+Navigate [TF-M](https://www.trustedfirmware.org/projects/tf-m/).
+Then go through **DOCS** → **Getting Started Guides** → **Software requirements** for TF-M build environment setup.
+
+**NOTE**: For unknown reason, **GNU Arm Embedded Toolchain 10-2020-q4-major** built code **FAILS** to run. Avoid this toolchain version.
```

Comment:
is this reported anywhere?

Reply:
No. The cause is unknown. The TF-M is built on **GNU Arm Embedded Toolchain 9-2020-q2-update**.

Reply:
@0xc0170 Unfortunately we need to avoid 10-2020-q4-major, see https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/9284.
@ccli8 Does it looks related?

Reply:
Actually, we didn't get to the root cause 10-2020-q4-major failing. Just cross-check and do no gcc compiler upgrade to this version.